### DISCUSSION TITLE
Custom element statuses

### DISCUSSION THREAD

**Discussion by angrybrad** [craftcms/cms#8127] [Ideas]
> Created by: Russ Back (russ@rallyagency.co.uk) on 2015/03/24 08:31:58 +0000
> Votes at time of UserVoice import: 20

-----------------------------------

Creating custom statuses would enable users to move entries from Draft to Ready for Review to Live etc.

The only way I can see of handling workflow like this at present is to create a field for an entry, rather than a custom system status.

**Comment by angrybrad** [comment]
> Posted by Michael (michael@michaelrog.com) on 2015/03/11 15:06:07 +0000
-----------------------------------

I actually like the existing/restricted functionality of Statuses and would prefer to use custom fields for custom use cases. (Status info isn&#39;t actually stored in the database, best I can tell — It&#39;s calculated dynamically based on the postDate.) Perhaps an equally useful solution would be to have the ability to re-translate the status names on a per-section basis and/or to re-translate the label of the Post Date field.

**Comment by angrybrad** [comment]
> Posted by Johannes Lamers (johanneslamers@gmail.com) on 2015/03/10 17:03:25 +0000
-----------------------------------

I like both ideas. Custom fields to the right panel would be perfect!

**Comment by angrybrad** [comment]
> Posted by Stephen McIver (spm999@gmail.com) on 2015/04/13 23:58:09 +0000
-----------------------------------

I&#39;d also like the ability to create custom statuses. Sometimes, the client just wants to have more statuses than enabled and disabled, or have them renamed to something else that makes sense for their particular use case.

**Comment by angrybrad** [comment]
> Posted by Christian (seelbach@gmx.de) on 2015/03/09 00:49:13 +0000
-----------------------------------

Maybe it would also work to be able to add (custom) fields to the right panel?

**Comment by AbbeyDesign** [comment]
In EE, we were given 2-3 set statuses and then the ability to add custom ones. Freeform's Craft plugin has this ability now. Would really like this feature built into the core, it's a feature we missed from EE and installing Freeform only reminded us how indispensable this is.

**Comment by brandonkelly** [comment]
#1630 would help with this – then you could create a custom Dropdown field with your desired statuses, and create some Smart Sources for each of its possible values.

**Comment by carlcs** [comment]
I just ran into a situation where I could really benefit from overriding `Entry::getStatus()`. You think it’s reasonable to add new events for it @brandonkelly?

**Comment by brandonkelly** [comment]
I could see maybe allowing custom statues via an event.

**Comment by bymayo** [comment]
Similar, also need this for users so I can assign different user statuses. Any movement on this?

**Comment by philipboomy** [comment]
Building a real estate site where it would be nice to have custom status e.g. rented, sold, etc - Instead of having to build custom element type it would be nice to just use a channel and custom statues

**Reply by davorpeic** [reply]
You can do this easily with Categories, Custom dropdown field etc, no need to touch Entries status for this, I think it would be overkill for your case.
